### DISCUSSION TITLE
Thread logs created in airflow triggers

### DISCUSSION THREAD

**Discussion by swarnadeepC** [apache/airflow#36778] [Q&A]
Hi ,
I  am on airflow version 2.7.1
I am trying to submit few parallel tasks using an threadpoolexecutor in my airflow trigger. When the trigger runs the thread logs goes to airflow triggerer log while i expect it to the individual trigger logs  (https://github.com/apache/airflow/pull/27758) .Is my expectation correct ? Adding a sample trigger code with comments for further clarification.  

```
import time
from concurrent.futures import ThreadPoolExecutor
from typing import AsyncIterator
from airflow.triggers.base import BaseTrigger, TriggerEvent

class CustomTrigger(BaseTrigger):
    def __init__(self, dag_id, run_id):
        super().__init__()

        self.dag_id = dag_id
        self.run_id = run_id

    def serialize(self):
        return ("operators.CustomTrigger.CustomTrigger", {"dag_id": self.dag_id, "run_id":self.run_id})

    async def run(self) -> AsyncIterator[TriggerEvent]:
        self.log.info("Airflow custom triggerer is running") #-> these logs goes into the individual trigger logs 
        with ThreadPoolExecutor(max_workers=5) as executor:
            self.log.info('Start of job submission')
            results = executor.map(self.run_job, [1, 2, 3, 4, 5, 6])
            for r in results :
                self.log.info("finishing" + str(r))
        yield TriggerEvent("Complete")

    def run_job(self, number):

        self.log.info(" Run job for number for"+str(number)+" "+self.run_id) #-> These logs goes into airflow triggerer logs
        time.sleep(15)
        self.log.info(" Run job for finished for "+str(number)+" "+self.dag_id)
        return (number*number*number)
```

**Comment by potiuk** [comment]
I don't think it's a good idea to run stuff the way you run it. You should probably read and understand more how asyncio works, but generally your async method should never do a blocking call and clearly getting results from threads spawned by thread executor is a blocking calll. You can likely do what you want to do via running thread pool exacutor with `loop.run_in_executor()` as described in https://docs.python.org/3/library/asyncio-dev.html#concurrency-and-multithreading but I am not as experience in asyncio, so that's as much of an advice I can give.
